plugins/modules/ec2_instance.py

```diff
@@ -1950,12 +1950,7 @@ def ensure_present(existing_matches, desired_module_state, current_count=None):
         for ins in instances:
             # Wait for instances to exist (don't check state)
             try:
-                AWSRetry.jittered_backoff(
-                    catch_extra_error_codes=["InvalidInstanceID.NotFound"],
-                )(client.describe_instance_status)(
-                    InstanceIds=[ins["InstanceId"]],
-                    IncludeAllInstances=True,
-                )
+                client.describe_instance_status(InstanceIds=[ins["InstanceId"]], IncludeAllInstances=True)
```

Comment:
Please leave this line as it was:

1. It's currently explicitly retrying on "InvalidInstanceID.NotFound" to deal with certain race conditions within AWS (it returns the ID, but if you quickly query that API about the ID it fails to find it)
2. Without adding `aws_retry=True` to the describe_instance_status arguments the decorator won't perform any retries.

In this case I'd leave the change you made at line 2159, but use the original code for this call, and add  `InvalidInstanceID.NotFound` to the catch_extra_error_codes parameter that was here.